### PR TITLE
feat: allow for non-local websocket configuration

### PR DESCRIPTION
**Pull request by chris-dickson**
This does lazy loading of the jupyter session and allows for a paramterized configuration of chatt/TGPT

### CODE REVIEW

**Review by mattprintz — APPROVED** [review]
Looks good

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TGPTResource.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	@ConfigProperty(name = "tgpt.baseUrl", defaultValue = "/chatty/")
+	String baseUrl;
+
+	@ConfigProperty(name = "tgpt.appUrl", defaultValue = "http://localhost:8078/chatty/")
+	String appUrl;
+
+	@ConfigProperty(name = "tgpt.wsUrl", defaultValue = "ws://localhost:8078/chatty_ws/")
+	String wsUrl;
+
+	@ConfigProperty(name = "tgpt.token", defaultValue = "89f73481102c46c0bc13b2998f9a4fce")
```
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TGPTResource.java:30]: Should this be in the clear?

File: packages/client/hmi-client/src/services/jupyter.ts [hunk 2]
```diff
@@ -121,28 +122,40 @@ export interface IJupyterMessage<T extends JupyterMessageType = JupyterMessageTy
 export declare type JupyterMessage = IJupyterMessage | messages.Message;
 
 // TODO: These settings should be pulled from the environment variables or appropriate config setup.
```
Comment (dvince2) [inline packages/client/hmi-client/src/services/jupyter.ts:124]: I think this TODO is now resolved

**Review by dvince2 — APPROVED** [review]

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TGPTResource.java (hunk 1 above)
Comment (chris-dickson) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TGPTResource.java:30]: This is only for local dev.  In production we will have a secret keypair.

**Pull request merged**